trade-finance-manager-ui/server/helpers/deal-cancellation-enabled.test.ts

```diff
@@ -0,0 +1,55 @@
+import * as ukefCommon from '@ukef/dtfs2-common';
+import { DEAL_SUBMISSION_TYPE } from '@ukef/dtfs2-common';
+import { dealCancellationEnabled } from './deal-cancellation-enabled';
+
+describe('dealCancellationEnabled', () => {
+  const isTfmDealCancellationFeatureFlagEnabledSpy = jest.spyOn(ukefCommon, 'isTfmDealCancellationFeatureFlagEnabled');
```

Comment:
I've actually not often seen spy be used for this -- Is there a reason to favour it? Personally, I think a `mock` is probably the right choice here as we're `mocking` it rather than `spying` on it, but keen to understand!

Reply:
No that makes sense - so I originally did my usual mocking, but it was causing all sorts of TS errors on my IDE, so I did a hunt around and found other test files mocking feature flags in TS files and found a lot of instances with `isTfmPaymentReconciliationFeatureFlagEnabledSpy` - so thought (in potentially in a moment of madness) that maybe we do things differently in TS tests... but I wasn't super convinced by what I was doing so glad you've flagged.

I think I'm concluding that my IDE needs some reconfig to stop it showing non-existent TS errors! :)

Reply:
Have switched it back to the usual mocking